### PR TITLE
feat: columns with nested filters

### PR DESCRIPTION
**Pull request by VladanSMG**
References https://smg-au.atlassian.net/browse/DM-2875

## Motivation and context

We need to make sure the layout is not shifted to different column when nested checkboxes are expanded

## Before

Content from one column goes to the next column

## After

Content from one column remains in one column and only expands in height

## How to test

Check all the filters, everything should remain the same

### CODE REVIEW

File: src/components/checkboxFilter/index.tsx [hunk 1]
```diff
@@ -18,34 +18,55 @@ function CheckboxFilter<ItemKey extends string, FilterName extends string>({
...
+  );
+
   return (
     <TranslationProvider language={language} scopes={['checkboxFilter']}>
       <Box
         sx={{
-          columns: { md: numberOfColumnsOnDesktop, base: 1 },
-          columnRule: 'solid var(--chakra-colors-gray-100) 1px',
-          columnGap: 'var(--chakra-space-4xl)',
+          display: 'grid',
```
Comment (talamcol) [inline src/components/checkboxFilter/index.tsx:41]: I think you accidentally missed to add the gray separator we had before `columnRule: 'solid var(--chakra-colors-gray-100) 1px',`

File: src/components/checkboxFilter/index.tsx [hunk 2]
```diff
@@ -18,34 +18,55 @@ function CheckboxFilter<ItemKey extends string, FilterName extends string>({
...
+        items.length,
+      );
+      return [...items].slice(columnStartIndex, columnEndIndex);
+    },
+  );
+
   return (
     <TranslationProvider language={language} scopes={['checkboxFilter']}>
       <Box
         sx={{
```
Comment (talamcol) [inline src/components/checkboxFilter/index.tsx:40]: `sx` is only supposed to be used if you have custom css that chakra does not support. Here you can add the `Grid` component

File: src/components/checkboxFilter/index.tsx [hunk 3]
```diff
@@ -18,34 +18,55 @@ function CheckboxFilter<ItemKey extends string, FilterName extends string>({
...
-              item={item}
-              onApply={onApply}
-              indentFacet={hasGroups}
-            />
-          );
-        })}
+        {columns.map((columnItems, columnIndex) => (
+          <div key={columnIndex} data-testid="column">
+            {columnItems.map((item) => (
+              <div key={item.key} data-testid={`item-${item.key}`}>
```
Comment (talamcol) [inline src/components/checkboxFilter/index.tsx:52]: would it also work without the additional div or has it a purpose besides having a test-id?

**Review by talamcol — COMMENTED** [review]
Could we please open a PR in listings-web with the pre-release? Thank you 🙏

File: src/components/checkboxFilter/index.tsx (hunk 3 above)
Comment (VladanSMG) [reply src/components/checkboxFilter/index.tsx]: I think it would work, but I wanted a test to be there for easier queries and to be covered.

**Pull request closed without merging**